lib/verify.js

```diff
@@ -9,48 +9,50 @@ var crypto = require('crypto');
 
 module.exports = {
   /**
-   * Verify RSA/DSA signature against public key.  You are expected to pass in
-   * an object that was returned from `parse()`.
+   * Simply wraps up the node crypto operations for you, and returns
+   * true or false.  You are expected to pass in an object that was
+   * returned from `parse()`.
    *
    * @param {Object} parsedSignature the object you got from `parse`.
-   * @param {String} pubkey RSA/DSA private key PEM.
+   * @param {String} key either an RSA private key PEM or HMAC secret.
    * @return {Boolean} true if valid, false otherwise.
    * @throws {TypeError} if you pass in bad arguments.
    */
-  verifySignature: function verifySignature(parsedSignature, pubkey) {
+  verifySignature: function verifySignature(parsedSignature, key) {
     assert.object(parsedSignature, 'parsedSignature');
-    assert.string(pubkey, 'pubkey');
+    assert.string(key, 'key');
 
-    var alg = parsedSignature.algorithm.match(/^(RSA|DSA)-(\w+)/);
+    var alg = parsedSignature.algorithm.match(/(HMAC|RSA|DSA)-(\w+)/);
     if (!alg || alg.length !== 3)
       throw new TypeError('parsedSignature: unsupported algorithm ' +
                           parsedSignature.algorithm);
 
-    var verify = crypto.createVerify(alg[0]);
-    verify.update(parsedSignature.signingString);
-    return verify.verify(pubkey, parsedSignature.params.signature, 'base64');
+    if (alg[1] === 'HMAC') {
+      var hmac = crypto.createHmac(alg[2].toUpperCase(), key);
+      hmac.update(parsedSignature.signingString);
+      return (hmac.digest('base64') === parsedSignature.params.signature);
```

Comment:
Should do a [double-hmac comparison](https://www.nccgroup.trust/us/about-us/newsroom-and-events/blog/2011/february/double-hmac-verification/) to protect against timing attacks.